### PR TITLE
feat(cnx-help): Fix tests for elements V1

### PR DESCRIPTION
**Pull request by umahaea**
* Write the styling tests for contextual-help that points to elements sdk V1. This is for both desktop and  mobile
* Remove the existing V0 specific styling tests
* Fix the pageobjects for changed identifiers
* Copy 'icons' directory to root during npm install of contextual-help

@StommePoes @wyseguyonline @JasonLantz @mhomolak for your review.

### CODE REVIEW

File: src/test/java/origamiV2Tests/ContextualHelpTest.java [hunk 1]
```diff
@@ -595,22 +567,241 @@ private void trapFocusWithinDrawerInHelpTopicsDetailTest() {
...
+        commonUtils.getUrl(contextualHelpUrl);
+        commonUtils.click(appHeaderPgObj.clickableHelpLink);
+        paddingLeft = commonUtils.getCSSValue(conxHelpPgObj.contextualHelpDrawerHeader, "padding-left");
+        isPaddingLeft = commonUtils.assertValue(paddingLeft, "32px", " padding-left to contextual-help-drawer header is not as per the spec");
+        paddingRight = commonUtils.getCSSValue(conxHelpPgObj.contextualHelpDrawerHeader, "padding-right");
+        isPaddingRight = commonUtils.assertValue(paddingRight, "24px", " padding-right to contextual-help-drawer header is not as per the spec");
+        paddingTop = commonUtils.getCSSValue(conxHelpPgObj.contextualHelpDrawerHeader, "padding-top");
+        isPaddingTop = commonUtils.assertValue(paddingTop, "20px", " padding-top to contextual-help-drawer header is not as per the spec");
+        paddingBottom = commonUtils.getCSSValue(conxHelpPgObj.contextualHelpDrawerHeader, "padding-bottom");
+        isPaddingBottom = commonUtils.assertValue(paddingBottom, "20px", " padding-right to contextual-help-drawer header is not as per the spec");
```
Comment (StommePoes) [inline src/test/java/origamiV2Tests/ContextualHelpTest.java]: padding bottom or padding right?

File: src/test/java/origamiV2Tests/ContextualHelpTest.java [hunk 2]
```diff
@@ -595,22 +567,241 @@ private void trapFocusWithinDrawerInHelpTopicsDetailTest() {
...
+
+    @Test(testName = "Verify styles for SVG icon test", dataProvider = "Verify SVG icon style test data", groups = "desktop-regression")
+    private void styleForSVGTest(String icon, String expWidth, String expHeight) {
+        commonUtils.getUrl(contextualHelpUrl);
+        commonUtils.click(appHeaderPgObj.clickableHelpLink);
+        commonUtils.click(conxHelpPgObj.helpTopicTitle);
+        width = commonUtils.getCSSValue(conxHelpPgObj.backToHelpTopicsIcon, "width");
+        isWidth = commonUtils.assertValue(width, expWidth, " width of svg " + icon + " is not as per the spec");
+        height = commonUtils.getCSSValue(conxHelpPgObj.backToHelpTopicsIcon, "height");
+        isHeight = commonUtils.assertValue(height, expHeight, " height of svg " + icon + " is not as per the spec");
```
Comment (StommePoes) [inline src/test/java/origamiV2Tests/ContextualHelpTest.java]: Just so you know, these should pass even in IE-- the outer SVG works fine, just the visible icons inside won't show.

File: src/test/java/origamiV2Tests/ContextualHelpTest.java [hunk 3]
```diff
@@ -595,22 +567,241 @@ private void trapFocusWithinDrawerInHelpTopicsDetailTest() {
...
+        commonUtils.click(appHeaderPgObj.clickableHelpLink);
+        commonUtils.click(conxHelpPgObj.helpTopicTitle);
+        paddingLeft = commonUtils.getCSSValue(conxHelpPgObj.backToHelpTopicsButton, "padding-left");
+        isPaddingLeft = commonUtils.assertValue(paddingLeft, "0px", " padding-left to contextual-help-drawer back-to-help button is not as per the spec");
+        paddingRight = commonUtils.getCSSValue(conxHelpPgObj.backToHelpTopicsButton, "padding-right");
+        isPaddingRight = commonUtils.assertValue(paddingRight, "0px", " padding-right to contextual-help-drawer back-to-help button is not as per the spec");
+        paddingTop = commonUtils.getCSSValue(conxHelpPgObj.backToHelpTopicsButton, "padding-top");
+        isPaddingTop = commonUtils.assertValue(paddingTop, "0px", " padding-top to contextual-help-drawer back-to-help button is not as per the spec");
+        paddingBottom = commonUtils.getCSSValue(conxHelpPgObj.backToHelpTopicsButton, "padding-bottom");
+        isPaddingBottom = commonUtils.assertValue(paddingBottom, "0px", " padding-right to contextual-help-drawer back-to-help button is not as per the spec");
```
Comment (StommePoes) [inline src/test/java/origamiV2Tests/ContextualHelpTest.java]: padding bottom or padding right?

File: src/test/java/origamiV2Tests/ContextualHelpTest.java [hunk 4]
```diff
@@ -878,6 +1037,219 @@ private void openAndThenRemoveHelpTopicsMobileTest(String noOfTopics, String top
...
+        commonUtils.getUrl(contextualHelpUrl, "mobile");
+        commonUtils.click(appHeaderPgObj.clickableHelpLink, "mobile");
+        paddingLeft = commonUtils.getCSSValue(conxHelpPgObj.helpTopicLayout, "padding-left", "mobile");
+        isPaddingLeft = commonUtils.assertValue(paddingLeft, "32px", " padding-left to contextual-help topic is not as per the spec");
+        paddingRight = commonUtils.getCSSValue(conxHelpPgObj.helpTopicLayout, "padding-right", "mobile");
+        isPaddingRight = commonUtils.assertValue(paddingRight, "32px", " padding-right to contextual-help topic is not as per the spec");
+        paddingTop = commonUtils.getCSSValue(conxHelpPgObj.helpTopicLayout, "padding-top", "mobile");
+        isPaddingTop = commonUtils.assertValue(paddingTop, "0px", " padding-top to contextual-help topic is not as per the spec");
+        paddingBottom = commonUtils.getCSSValue(conxHelpPgObj.helpTopicLayout, "padding-bottom", "mobile");
+        isPaddingBottom = commonUtils.assertValue(paddingBottom, "0px", " padding-right to contextual-help topic is not as per the spec");
```
Comment (StommePoes) [inline src/test/java/origamiV2Tests/ContextualHelpTest.java]: padding bottom or padding right?

File: src/test/java/origamiV2Tests/ContextualHelpTest.java [hunk 5]
```diff
@@ -878,6 +1037,219 @@ private void openAndThenRemoveHelpTopicsMobileTest(String noOfTopics, String top
...
+        commonUtils.getUrl(contextualHelpUrl, "mobile");
+        commonUtils.click(appHeaderPgObj.clickableHelpLink, "mobile");
+        paddingLeft = commonUtils.getCSSValue(conxHelpPgObj.contextualHelpDrawerHeader, "padding-left", "mobile");
+        isPaddingLeft = commonUtils.assertValue(paddingLeft, "32px", " padding-left to contextual-help-drawer header is not as per the spec");
+        paddingRight = commonUtils.getCSSValue(conxHelpPgObj.contextualHelpDrawerHeader, "padding-right", "mobile");
+        isPaddingRight = commonUtils.assertValue(paddingRight, "24px", " padding-right to contextual-help-drawer header is not as per the spec");
+        paddingTop = commonUtils.getCSSValue(conxHelpPgObj.contextualHelpDrawerHeader, "padding-top", "mobile");
+        isPaddingTop = commonUtils.assertValue(paddingTop, "20px", " padding-top to contextual-help-drawer header is not as per the spec");
+        paddingBottom = commonUtils.getCSSValue(conxHelpPgObj.contextualHelpDrawerHeader, "padding-bottom", "mobile");
+        isPaddingBottom = commonUtils.assertValue(paddingBottom, "20px", " padding-right to contextual-help-drawer header is not as per the spec");
```
Comment (StommePoes) [inline src/test/java/origamiV2Tests/ContextualHelpTest.java]: padding bottom or padding right?

File: src/test/java/origamiV2Tests/ContextualHelpTest.java [hunk 6]
```diff
@@ -878,6 +1037,219 @@ private void openAndThenRemoveHelpTopicsMobileTest(String noOfTopics, String top
...
+        commonUtils.click(conxHelpPgObj.helpTopicTitle, "mobile");
+        paddingLeft = commonUtils.getCSSValue(conxHelpPgObj.backToHelpTopicsButton, "padding-left", "mobile");
+        isPaddingLeft = commonUtils.assertValue(paddingLeft, "0px", " padding-left to contextual-help-drawer back-to-help button is not as per the spec");
+        paddingRight = commonUtils.getCSSValue(conxHelpPgObj.backToHelpTopicsButton, "padding-right", "mobile");
+        isPaddingRight = commonUtils.assertValue(paddingRight, "0px", " padding-right to contextual-help-drawer back-to-help button is not as per the spec");
+        paddingTop = commonUtils.getCSSValue(conxHelpPgObj.backToHelpTopicsButton, "padding-top", "mobile");
+        isPaddingTop = commonUtils.assertValue(paddingTop, "0px", " padding-top to contextual-help-drawer back-to-help button is not as per the spec");
+        paddingBottom = commonUtils.getCSSValue(conxHelpPgObj.backToHelpTopicsButton, "padding-bottom", "mobile");
+        isPaddingBottom = commonUtils.assertValue(paddingBottom, "0px", " padding-right to contextual-help-drawer back-to-help button is not as per the spec");
```
Comment (StommePoes) [inline src/test/java/origamiV2Tests/ContextualHelpTest.java]: padding bottom, padding right

**Comment by StommePoes** [comment]
All my comments inline.

Further looks good, except I'm not 100% certain if Design wants the app-header to remain on the page when the Drawer/Help is open, or if Help is meant to cover the entire page (we should discuss with them). Position: fixed works crappy on many mobile browsers and we might want to remove all that at the smallest responsive size.

File: src/main/java/origamiV2/jsfiles/contextualHelp/contextual-help.js [hunk 7]
```diff
@@ -41,25 +41,25 @@ document.openThenRemoveTopics = function(topics, delay){
...
...ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p> </div> <h3><button class="o-disclosure" aria-controls="item-4" aria-expanded="false"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" class="pe-icon--pivot-close-18"><use xlink:href="/icons/p-icons-sprite-1.1.svg#pivot-close-18"></use></svg> Item 4</button></h3> <div id="item-4" class="o-panel--closed"> <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p> </div> </div>'
```
Comment (umahaea) [inline src/main/java/origamiV2/jsfiles/contextualHelp/contextual-help.js]: Ok..

File: src/main/shell_scripts/components.sh [hunk 8]
```diff
@@ -28,14 +28,14 @@ cp ~/build/Pearson-Higher-Ed/ux-test-platform/app-header/node_modules/pearson-el
...
-echo -e "******************************\\n    Installing contextual-help    \\n******************************"
+echo -e "******************************\\n    Installing contextual-help V2   \\n******************************"
 git clone https://github.com/Pearson-Higher-Ed/contextual-help.git
 cd contextual-help
 git checkout $1
 npm install &>/dev/null
 npm run build &>/dev/null
 cp ~/build/Pearson-Higher-Ed/ux-test-platform/contextual-help/build/dist.contextual-help.js ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/origamiV2/jsfiles/contextualHelp/
-cp -R ~/build/Pearson-Higher-Ed/ux-test-platform/contextual-help/node_modules/pearson-elements/dist/fonts ~/build/Pearson-Higher-Ed/ux-test-platform/
+cp -R ~/build/Pearson-Higher-Ed/ux-test-platform/contextual-help/node_modules/pearson-elements/dist/icons ~/build/Pearson-Higher-Ed/ux-test-platform/
```
Comment (umahaea) [inline src/main/shell_scripts/components.sh]: Ok..

File: src/test/java/origamiV2Tests/ContextualHelpTest.java (hunk 1 above)
Comment (umahaea) [inline src/test/java/origamiV2Tests/ContextualHelpTest.java]: sorry typo..padding bottom..

File: src/test/java/origamiV2Tests/ContextualHelpTest.java (hunk 2 above)
Comment (umahaea) [inline src/test/java/origamiV2Tests/ContextualHelpTest.java]: OK..

File: src/test/java/origamiV2Tests/ContextualHelpTest.java (hunk 3 above)
Comment (umahaea) [inline src/test/java/origamiV2Tests/ContextualHelpTest.java]: sorry typo..padding bottom..

File: src/test/java/origamiV2Tests/ContextualHelpTest.java (hunk 4 above)
Comment (umahaea) [inline src/test/java/origamiV2Tests/ContextualHelpTest.java]: sorry typo..padding bottom..

File: src/test/java/origamiV2Tests/ContextualHelpTest.java (hunk 6 above)
Comment (umahaea) [inline src/test/java/origamiV2Tests/ContextualHelpTest.java]: ok..

File: src/test/java/origamiV2Tests/ContextualHelpTest.java (hunk 5 above)
Comment (umahaea) [inline src/test/java/origamiV2Tests/ContextualHelpTest.java]: sorry typo..padding bottom..

File: src/main/java/origamiV2/origamiV2PageObjects/ContextualHelpPageObjects.java [hunk 9]
```diff
@@ -42,13 +48,15 @@ public String xpathForHelpTopicExcerpt(String item, int x){
...
     }
 
-    public By helpContentTopicDetailVisible=By.xpath("//div[@class='o-drawer-right o-drawer-animated o-contextual-help__drawer o-drawer o-drawer-open o-contextual-help__content--visible']");
-    public By helpContentTopicDetailReopen=By.xpath("//div[@class='o-drawer-right o-drawer-animated o-contextual-help__drawer o-drawer o-contextual-help__content--visible o-drawer-open']");
-    public By helpContentTopicDetailHidden=By.xpath("//div[@class= 'o-drawer-right o-drawer-animated o-contextual-help__drawer o-drawer o-contextual-help__content--visible']");
+    public By helpContentTopicDetailVisible=By.xpath("//div[@class='o-drawer-right o-drawer-animated o-contextual-help__drawer exists-o-header o-drawer o-drawer-open o-contextual-help__content--visible']");
+    public By helpContentTopicDetailReopen=By.xpath("//div[@class='o-drawer-right o-drawer-animated o-contextual-help__drawer exists-o-header o-drawer o-contextual-help__content--visible o-drawer-open']");
+    public By helpContentTopicDetailHidden=By.xpath("//div[@class= 'o-drawer-right o-drawer-animated o-contextual-help__drawer exists-o-header o-drawer o-contextual-help__content--visible']");
```
Comment (umahaea) [inline src/main/java/origamiV2/origamiV2PageObjects/ContextualHelpPageObjects.java]: added test to test 2 sepearate instances with and without app-header And also verify the css top-padding for with and without app-header.

**Pull request merged**